src/WebJobs.Extensions.DurableTask/MessagePayloadDataConverter.cs

```diff
@@ -13,31 +13,68 @@ namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
 {
     internal class MessagePayloadDataConverter : JsonDataConverter
     {
-        // The default JsonDataConverter for DTFx includes type information in JSON objects. This causes issues
-        // because the type information generated from C# scripts cannot be understood by DTFx. For this reason, explicitly
-        // configure the JsonDataConverter to not include CLR type information. This is also safer from a security perspective.
-        internal static readonly JsonSerializerSettings MessageSettings = new JsonSerializerSettings
-        {
-            TypeNameHandling = TypeNameHandling.None,
-        };
-
         private static readonly JsonSerializerSettings ErrorSettings = new JsonSerializerSettings

```

Comment:
Since we only use this when the customer doesn't supply their own, let's only initialize it inside of the new `ErrorConverter` property you added below.